Example/AsyncReactorExample/Features/Repository/Search/RepositorySearchView.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  RepositorySearchView.swift
+//  AsyncReactorExample
+//
+//  Created by Ahmet Bozkan on 22.05.23.
+//
+
+import SwiftUI
+import AsyncReactor
+import Combine
+
+struct RepositorySearchView: View {
+    @EnvironmentObject
+    private var reactor: RepositorySearchReactor
+    
+    @ActionBinding(RepositorySearchReactor.self, keyPath: \.hidePrivate, action: RepositorySearchReactor.Action.onHidePrivateToggle)
+    private var hidePrivate: Bool
+    
+    @ActionBinding(RepositorySearchReactor.self, keyPath: \.query, action: RepositorySearchReactor.Action.enterQuery)
+    private var query: String
+    
+    private let queryPublisher = PassthroughSubject<String, Never>()
+    
+    var body: some View {
+        NavigationView {
+            List {
+                Toggle("Hide Private Repos", isOn: $hidePrivate)
+                
+                if reactor.isLoading {
+                    HStack {
+                        Spacer()
+                        ProgressView().id(UUID())
+                        Spacer()
+                    }
+                    .padding()
+                }
+                
+                if !reactor.repositories.isEmpty {
+                    RepositoryList(
+                        repositories: reactor.repositories.filter { !hidePrivate || $0.isVisible }
+                    )
+                }
+            }
+            .navigationTitle("Repositories")
+            .refreshable {
+                reactor.send(.load)
+            }
+            
+            .searchable(text: $query)
+            .onChange(of: query) { query in
+                queryPublisher.send(query)
+            }
+            .onReceive(
+                queryPublisher
```

Comment:
You can and should do the throttle / debounce in the reactor. The idea is to use a cancelId with mode `inFlight` for the query binding. Then it will cancel the previous search action, if one is already running. In the reactor you currently have `logger.debug("search: \(query)")`, you should trigger the search request.
Here in the view you should only need the .searchable modifier. No need for the query publisher and such